[EXPERIMENTAL] Make nil/true/false to_s return a frozen string

casperisfine opened the pull request:
Ref: https://bugs.ruby-lang.org/issues/16150?next_issue_id=16184
Ref: https://github.com/ruby/ruby/pull/2487
Ref: https://github.com/ruby/ruby/pull/2437

This is the last part of feature 16150.

cc @nobu @eregon

eregon commented on spec/ruby/core/true/to_s_spec.rb:
```diff
@@ -4,4 +4,10 @@
   end
+
+  ruby_version_is "2.7" do
+    it "returns a frozen string ''" do
```
`it "returns the frozen string 'true'" do`

eregon commented on spec/ruby/core/false/to_s_spec.rb:
```diff
@@ -4,4 +4,10 @@
   end
+
+  ruby_version_is "2.7" do
+    it "returns a frozen string ''" do
```
`it "returns the frozen string 'false'" do`

eregon commented on spec/ruby/core/false/to_s_spec.rb:
```diff
@@ -4,4 +4,10 @@
+  ruby_version_is "2.7" do
+    it "returns a frozen string ''" do
+      false.to_s.frozen?.should == true
+    end
```
Could you add a spec that it's always the same String returned, as for Module#name & Symbol#to_s?

eregon commented:
Looks good, @nobu can you review too?

casperisfine commented:
@eregon sorry for the bad copy paste mistakes, I just fixed them.

I also added the same instance test. 

Also not sure why GItHub Actions won't build :/

nobu commented on object.c:
```diff
@@ -40,6 +40,10 @@ VALUE rb_cNilClass; /*!< NilClass class */
 
+VALUE rb_cNilClass_to_s;
+VALUE rb_cTrueClass_to_s;
+VALUE rb_cFalseClass_to_s;
```
Should be `static` if no specific reason.

casperisfine replied:
Done.

eregon commented:
I'm not sure why GitHub Actions don't trigger, maybe try amending the commit/rebase and push? cc @hsbt 
Could you maybe make sure `make test-all TESTS=-j8` and `make test-spec MSPECOPT=-j` pass locally?

eregon commented:
I sent an email to GitHub Support since that's what https://github.com/ruby/ruby/pull/2490/checks?check_run_id=237695456 says to do.

eregon commented:
@casperisfine it seems the issue with GitHub Actions not running is that the PR is from a fork of `github/ruby` instead of `ruby/ruby`.

byroot commented:
@eregon ah! Not surprised. I meant to fix that a few weeks ago, but never did.

Do you want me to re-open another PR to get a proper CI run?

eregon commented:
@byroot It's probably the easiest way, yes (sorry about that).
I'd like to see the CI run before merging :)

byroot commented:
No worries, and not your fault.

casperisfine commented:
Closing in favor of https://github.com/ruby/ruby/pull/2492

eregon commented:
Here is what GitHub support replied:

> Thank you for reaching out to GitHub Support!
> 
> Our Engineers recently deployed a fix for this, could you give it another try and let me know if you encounter any other issues?

So, it should work next time :)

byroot commented:
Thanks. I actually reforked directly from `ruby/ruby`, so that's fine. But it's a good thing they fixed it.
